.github/workflows/branch-suggestion.yml

```diff
@@ -0,0 +1,49 @@
+name: Branch Suggestion for PRs
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  workflow_call:
+    secrets:
+      JIRA_TOKEN:
+        required: true
+        description: 'Pre-encoded JIRA token (base64(email:api_token))'
+
+permissions:
+  pull-requests: write
+  contents: read
+
+jobs:
+  suggest-branches:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout github-actions repository
+        uses: actions/checkout@v4
+        with:
+          repository: TykTechnologies/github-actions
+          ref: main
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+
+      - name: Install dependencies
+        working-directory: branch-suggestion
+        run: npm install
+
+      - name: Install Visor
+        run: npm install -g @probelabs/visor
```

Comment:
shall we make this a docker based github action with node and visor preinstalled?